src/exporters/http/retry.rs

```diff
@@ -105,10 +105,11 @@ where
         _req: &mut Request<ReqBody>,
         result: &mut Result<Response<Resp>, BoxError>,
     ) -> Option<Self::Future> {
-        // Should never happen
+        // Set the request start time. Ideally we would know the start time from the start of the
+        // execution of the request, but we don't have that data in the request yet. Instead,
+        // the actual max elasped time may include the initial timeout. TODO
         if self.request_start.is_none() {
```

Comment:
So did this actually happen or is it a change because we were failing to retry under some scenario?

Reply:
yeah, this was found from exploration of https://github.com/streamfold/rotel/issues/135